protocols/kad/src/behaviour.rs

```diff
@@ -174,24 +171,6 @@ pub struct KademliaConfig {
     provider_publication_interval: Option<Duration>,
     connection_idle_timeout: Duration,
     kbucket_inserts: KademliaBucketInserts,
-    caching: KademliaCaching,
-}
-
-/// The configuration for Kademlia "write-back" caching after successful
-/// lookups via [`Kademlia::get_record`].
-#[derive(Debug, Clone)]
-pub enum KademliaCaching {
-    /// Caching is disabled and the peers closest to records being looked up
-    /// that do not return a record are not tracked, i.e.
-    /// [`GetRecordOk::cache_candidates`] is always empty.
-    Disabled,
-    /// Up to `max_peers` peers not returning a record that are closest to the key
-    /// being looked up are tracked and returned in [`GetRecordOk::cache_candidates`].
-    /// Furthermore, if [`Kademlia::get_record`] is used with a quorum of 1, the
-    /// found record is automatically sent to (i.e. cached at) these peers. For lookups with a
-    /// quorum > 1, the write-back operation must be performed explicitly, if
-    /// desired and after choosing a record from the results, via [`Kademlia::put_record_to`].
-    Enabled { max_peers: u16 },
```

Comment:
With caching removed, `PutRecordContext::Cache` needs to be removed as well.

Reply:
done